### PR TITLE
More TCP metrics support

### PR DESCRIPTION
**Pull request by howardjohn**
* Add metrics with splice
* Give telemetry at all points we write
* Test and validate the telemetry

With this, Kiali shows the mesh with just ztunnel.

Not yet implemented is looking up source attributes for baggage or XDS, but can be a followup

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -59,7 +59,7 @@ tower = { version = "0.4.12", features = ["full"] }
 tracing = "0.1.34"
 tracing-subscriber = { version = "0.3.16" , features = ["registry", "env-filter"]}
 dyn-clone = "1.0.9"
-realm_io = "0.3.5"
+realm_io = { git = "https://github.com/howardjohn/realm", branch = "io/copy-bytes-sent" }
```
Comment (hzxuzhonghu) [inline Cargo.toml:62]: Will you file a pr to realm?

Comment (howardjohn) [reply Cargo.toml:62]: https://github.com/zhboner/realm/pull/95. We can see what they say before we merge this, hopefully they are ok with it

Comment (howardjohn) [reply Cargo.toml:62]: Merged, switched to real dep

File: src/test_helpers/tcp.rs [hunk 2]
```diff
@@ -121,11 +127,28 @@ where
...
+            loop {
+                let read = r.read(&mut buffer).await.expect("tcp ready");
+                if read == 0 {
+                    break;
+                }
+                let wrote = w.write(&buffer[..read]).await.expect("tcp ready");
+                if wrote == 0 {
+                    break;
+                }
+                let wrote = w.write(&buffer[..read]).await.expect("tcp ready");
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply src/test_helpers/tcp.rs:141]: this is is just for testing so we can make sure we record the "read" and "wrote" metrics correctly. Before we always read and write same data so we cannot actually test the 2 metrics we're not swapped accidentally

File: tests/namespaced.rs [hunk 3]
```diff
@@ -136,16 +170,70 @@ async fn test_hbone_request() -> anyhow::Result<()> {
...
+fn source_labels() -> HashMap<String, String> {
+    HashMap::from([("reporter".to_string(), "source".to_string())])
+}
+
+async fn verify_metrics(
+    ztunnel: TestApp,
+    assertions: &[(&str, u64)],
+    labels: &HashMap<String, String>,
+) {
+    // Wait form metrics to populate...
```
Comment (ymesika) [inline tests/namespaced.rs:204]: for

**Review by ymesika — APPROVED** [review]
Nice. lgtm.

**Pull request merged**